dev/merge_spark_pr.py

```diff
@@ -280,14 +271,30 @@ def resolve_jira_issue(merge_branches, comment, default_jira_id=""):
     )
 
     versions = asf_jira.project_versions("SPARK")
+    # Consider only x.y.z, unreleased, unarchived versions
+    versions = [
+        x
+        for x in versions
+        if not x.raw["released"] and not x.raw["archived"] and re.match(r"\d+\.\d+\.\d+", x.name)
+    ]
     versions = sorted(versions, key=lambda x: x.name, reverse=True)
-    versions = list(filter(lambda x: x.raw["released"] is False, versions))
-    # Consider only x.y.z versions
-    versions = list(filter(lambda x: re.match(r"\d+\.\d+\.\d+", x.name), versions))
 
-    default_fix_versions = list(
-        map(lambda x: fix_version_from_branch(x, versions).name, merge_branches)
-    )
+    default_fix_versions = []
```

Comment:
Actually seems this breaks the script:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/jira/resources.py", line 222, in __getattr__
    return self[item]  # type: ignore
TypeError: 'Version' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.../spark/./dev/merge_spark_pr.py", line 606, in <module>
    main()
  File "/.../spark/./dev/merge_spark_pr.py", line 590, in main
    resolve_jira_issues(title, merged_refs, jira_comment)
  File "/.../spark/./dev/merge_spark_pr.py", line 397, in resolve_jira_issues
    resolve_jira_issue(merge_branches, comment, jira_id)
  File "/.../spark/./dev/merge_spark_pr.py", line 303, in resolve_jira_issue
    (major, minor, patch) = v.split(".")
  File "/usr/local/lib/python3.9/site-packages/jira/resources.py", line 227, in __getattr__
    raise AttributeError(
AttributeError: <class 'jira.resources.Version'> object has no attribute 'split' ('Version' object is not subscriptable)
```

The reason seems that `default_fix_versions` has to be `Version` instance, and you should get the version from it via `.name`.

I am reverting this for now.

Reply:
Next time please use the updated merge script to merge the merge script update PR...

Reply:
Oh, ya, it's my bad. Sorry for the trouble, @HyukjinKwon and @cloud-fan .

Reply:
This should be my fault, I didn't validate the code below sufficiently.